daffodil-japi/src/main/java/org/apache/daffodil/japi/package-info.java

```diff
@@ -38,44 +38,48 @@
  * contain information about the parse/unparse, such as whether or not the
  * processing succeeded with any diagnostic information.
  *
- * The {@link org.apache.daffodil.japi.DataProcessor} also provides a function to create a
- * {@link org.apache.daffodil.japi.DaffodilXMLReader} that can be used to perform parsing via the
- * SAX API.
+ * The {@link org.apache.daffodil.japi.DataProcessor} also provides two functions that can be used to
+ * perform parsing/unparsing via the SAX API. The first creates a
+ * {@link org.apache.daffodil.japi.DaffodilParseXMLReader} which is used for parsing, and the
+ * second creates a {@link org.apache.daffodil.japi.DaffodilUnparseContentHandler} which is used for
+ * unparsing.
  *
  * <pre>
  * {@code
- * DaffodilXMLReader xmlRdr = dp.newXMLReaderInstance();
+ * DaffodilParseXMLReader xmlReader = dp.newXMLReaderInstance();
+ * DaffodilUnparseContentHandler unparseContentHandler = dp.newContentHandlerInstance(output);
  * }</pre>
  *
- * The {@link org.apache.daffodil.japi.DaffodilXMLReader} has several methods that allow one to
+ * The {@link org.apache.daffodil.japi.DaffodilParseXMLReader} has several methods that allow one to
  * set properties and handlers (such as ContentHandlers or ErrorHandlers) for the reader. One can
  * use any contentHandler/errorHandler as long as they extend the
  * {@link org.xml.sax.ContentHandler} and {@link org.xml.sax.ErrorHandler} interfaces
- * respectively. One can also set properties for the {@link org.apache.daffodil.japi.DaffodilXMLReader}
- * using {@link org.apache.daffodil.japi.DaffodilXMLReader#setProperty(java.lang.String, java.lang.Object)}.
+ * respectively. One can also set properties for the {@link org.apache.daffodil.japi.DaffodilParseXMLReader}
+ * using {@link org.apache.daffodil.japi.DaffodilParseXMLReader#setProperty(java.lang.String, java.lang.Object)}.
  *
  * The following properties can be set as follows:
+ *
+ * <p><i>The variables below resolve to "urn:ogf:dfdl:2013:imp:daffodil.apache.org:2018:sax:" and
+ * "BlobDirectory", "BlobPrefix" and "BlobSuffix" respectively.</i></p>
+ *
  * <pre>
  * {@code
- * xmlRdr.setProperty(XMLUtils.DAFFODIL_SAX_URN_BLOBDIRECTORY(), "/tmp/");
- * xmlRdr.setProperty(XMLUtils.DAFFODIL_SAX_URN_BLOBPREFIX(), "daffodil-sax-");
- * xmlRdr.setProperty(XMLUtils.DAFFODIL_SAX_URN_BLOBSUFFIX(), ".bin");
+ * xmlReader.setProperty(xmlReader.DAFFODIL_SAX_URN_BLOBDIRECTORY(),
+ *  Paths.get(System.getProperty("java.io.tmpdir"))); // value type: java.nio.file.Paths
+ * xmlReader.setProperty(xmlReader.DAFFODIL_SAX_URN_BLOBPREFIX(), "daffodil-sax-"); // value type String
+ * xmlReader.setProperty(xmlReader.DAFFODIL_SAX_URN_BLOBSUFFIX(), ".bin"); // value type String
```

Comment:
You're right. For some reason, intellij adds it in during autocomplete! Fixed!

Reply:
Ah I remember now why I did it like this! The code actually requires it. I'm not sure if it's because the api is implemented behind the scenes in scala. But I wanted to have the documentation match the expectation in the code, because the code won't compile without the (). So I've unfixed this, because if someone is trying to write code in Java using the API, they will need the parens

Reply:
Reference: https://alvinalexander.com/scala/how-to-access-scala-var-val-fields-from-java/

Reply:
Yeah, this is a downside with using Scala in Java. Much of what Scala generates in bytecode is actually public functions that get/set private variables, without a way to access the actual variables themselves. I'm not sure if there's away around this.